org.scala-ide.sdt.core/src/org/scalaide/ui/internal/editor/outline/ModelBuilder.scala

```diff
@@ -41,11 +41,16 @@ object ModelBuilder extends HasLogger {
     }
 
     def updateTree(parent: ContainerNode, t: Tree): Unit = {
+      def renderArgs[A](sb: StringBuilder, args: List[A])(f: A => Unit) = {
+        if (args.nonEmpty) {
+          f(args.head)
+          args.drop(1).foreach { a => sb.append(", "); f(a) }
```

Comment:
I agree with @wpopielarski that this could be simpler by replacing everything with something along the lines of

```
args.map(renderType(sb, _)).mkString(", ")
```